### ISSUE TITLE
BulkInsert on mssql

### ISSUE THREAD

**Issue by bassarisse** [sequelize/sequelize#4602]
I tried searching for a related issue, but I couldn't find, sorry if this is a duplicate!

When using `queryInterface.bulkInsert()` on mssql, an exception is thrown inside this method: https://github.com/sequelize/sequelize/blob/master/lib/dialects/mssql/query-generator.js#L210

It doesn't check if the `attributes` argument exists. Is anyone having the same problem?
I can try to fix it when I get some time, but maybe this is done by design? Not having the model attributes may cause undesirable effects, am I right?

Our current workaround is requiring the needed models and using `Model.bulkCreate()`.

PS: Sorry for not including the exception, I can't access the MSSQL database from the computer I'm using right now =(

**Comment by mickhansen** [comment]
The query interface is too tightly coupled in a lot of places i'm afraid.

**Comment by janmeier** [comment]
As far as I can see `attributes` are actually needed in some places, so I'd say its intentional. We don't document the API for the query interface since it's considered internal. I understand why you'd want to use it in a migration, but in this case I think you should just pass an empty array to 'fix it' :)

**Comment by mickhansen** [comment]
@janmeier Ideally we should start striving towards a place where the QueryInterface is as public and decoupled as possible.

**Comment by bassarisse** [comment]
Well, it is documented, at least partially, right? =) http://docs.sequelizejs.com/en/latest/docs/migrations/

Also, passing an empty object/array still throws an exception.
It would be trivial to check for undefined values, I'll just have have to see if there would be any side-effects.

**Comment by akanieski** [comment]
What I did for my migrations to support bulkInsert with MSSQL was passed in the attributes from the model. (ie. User.attributes).

Only downside is your migration needs to load up model definitions. 

I also noticed there is a flag in the options called `returning` which when flipped returns the results in array form. This is awesome but it doesn't match the results from SQLITE. With sqlite it returns the PK.

In my scenario the devs often do quick development against local sqlite and then move to mssql in test/qa environment, so supporting both db options is needed.
